utils/towers.ahk

```diff
@@ -104,3 +104,30 @@ GetRandom(ground_position, water_position) {
     }
     return [tower, ground_position]
 }
+
+Upgrade(tower, topCount, middleCount := 0, bottomCount := 0) {
+    if defeated {
+        return
+    }
+    x := TS[tower][2][1], y := TS[tower][2][2]
+
+    Click(x,y)          ; Open Tower
+    Sleep(100)
+    Loop topCount {
+		WaitForUpgrade1()
+        Send(KEYS["upgrade_1"])
+        Sleep(200)
+    }
+    Loop middleCount {
+		WaitForUpgrade2()
+        Send(KEYS["upgrade_2"])
+        Sleep(100)
+    }
+    Loop bottomCount {
+		WaitForUpgrade3()
+        Send(KEYS["upgrade_3"])
+        Sleep(100)
+    }
+    Send("{Esc}")       ; Close Tower
+    Sleep(200)
+}
```

Comment:
This could just be an option in the original function

Reply:
I messed up on this one, both forgetting to change the function name to UpgradeASAP and forgetting to increase sleep to 200. Will Edit this one, though I'm not sure how to make it an option in the original function.

Reply:
done